lib/sendf.c

```diff
@@ -328,9 +328,13 @@ CURLcode Curl_write(struct connectdata *conn,
   bytes_written = conn->send[num](conn, num, mem, len, &result);
 
   *written = bytes_written;
-  if(bytes_written >= 0)
+  if(bytes_written >= 0) {
+    /* if local port was not available on first send (TFO), update conninfo */
+    if(num == 0 && (conn->primary_port != 0) && (conn->local_port == 0))
+      Curl_updateconninfo(conn, conn->sock[FIRSTSOCKET]);
```

Comment:
`Curl_updateconninfo()` does nothing for `SOCK_DGRAM` sockets (UDP), so it'll keep calling that function during all UDP transfers. Not a big deal but also not elegant.

`Curl_updateconninfo()` has a check that only updates those port numbers if **not** doing TFO, which seems to void this entire approach? Shouldn't that condition also be modified?

Reply:
regarding the second point iirc tfo is actually disabled once it is sent, therefore eventually it will be false and conninfo will be updated. anyway i think maybe we should change that so tfo stays true if the connection was established with tfo and have a new bool like tfo_sent or something and then he could do a if(num ==0 && tfo_sent) do update etc

Reply:
hm reviewing the related threads just now I see there was a needs_update bit proposed in #1335, why was that abandoned?

Reply:
Yeah, I would prefer that. It feels a bit counter-intuitive to switch off the boolean that says TFO is requested after it has been "used". I also think we might soon find reasons for us to keep that knowledge around even after the initial data has been sent - for handling request replays properly/differently.